src/coreclr/interpreter/intrinsics.cpp

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "interpreter.h"
+#include "intrinsics.h"
+
+#define HAS_PREFIX(haystack, needle) \
+    (strncmp(haystack, needle, strlen(needle)) == 0)
+
+NamedIntrinsic GetNamedIntrinsic(COMP_HANDLE compHnd, CORINFO_METHOD_HANDLE compMethod, CORINFO_METHOD_HANDLE method)
+{
+    const char* className = NULL;
+    const char* namespaceName = NULL;
+    const char* methodName = compHnd->getMethodNameFromMetadata(method, &className, &namespaceName, NULL, 0);
+
+    // Array methods don't have metadata
+    if (!namespaceName)
+        return NI_Illegal;
+
+    if (!HAS_PREFIX(namespaceName, "System"))
+        return NI_Illegal;
+
+    if (!strcmp(namespaceName, "System"))
+    {
+        if (!strcmp(className, "Double") || !strcmp(className, "Single"))
+        {
+            if (!strcmp(methodName, "ConvertToIntegerNative"))
+                return NI_PRIMITIVE_ConvertToIntegerNative;
+            else if (!strcmp(methodName, "MultiplyAddEstimate"))
+                return NI_System_Math_MultiplyAddEstimate;
+        }
+        else if (!strcmp(className, "Debugger"))
+        {
+            // Interpreter-FIXME: No NI_ enum value for this?
+            if (!strcmp(methodName, "Break"))
+                return NI_Illegal;
+        }
+        else if (!strcmp(className, "Math") || !strcmp(className, "MathF"))
+        {
+            if (!strcmp(methodName, "ReciprocalEstimate"))
+                return NI_System_Math_ReciprocalEstimate;
```

Comment:
Should this be handled by modifying the managed code instead as we have discussed?

Reply:
Hmm, this is not the only math intrinsic that needs interpreting. I guess we can interpret this one too.